iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/mempool_introspection.inl

```diff
@@ -136,24 +139,28 @@ void MemPoolIntrospection<MemoryManager, SegmentManager, SenderPort>::send()
         auto sample = static_cast<Topic*>(chunkHeader->payload());
         new (sample) Topic;
 
-        prepareIntrospectionSample(
-            sample, posix::PosixGroup::getGroupOfCurrentProcess(), posix::PosixGroup::getGroupOfCurrentProcess(), id);
-        copyMemPoolInfo(*m_rouDiInternalMemoryManager, sample->m_mempoolInfo);
+        MemPoolIntrospectionInfo memPoolIntrospectionInfo;
```

Comment:
this can be optimized by using
```
if(sample->emplace_back())
{
    auto& memPoolIntrospectionInfo = sample->back();
    ...
}
```
This removes the copy at `sample->push_back(...);` and it could also be checked if the container is already full.